src/dbus/message.c

```diff
@@ -598,3 +599,52 @@ void message_stitch_sender(Message *message, uint64_t sender_id) {
         else
                 message->header->n_fields = htole32(message->n_header - sizeof(*message->header));
 }
+
+void message_log_append(Message *message, Log *log) {
+        log_appendf(log,
+                    "MESSAGE_DESTINATION=%s\n"
+                    "MESSAGE_SERIAL=%d\n"
+                    "MESSAGE_SIGNATURE=%s\n",
+                    message->metadata.fields.destination ?: "<broadcast>",
+                    message->metadata.header.serial,
+                    message->metadata.fields.signature ?: "<missing>");
+
+        switch (message->metadata.header.type) {
+        case DBUS_MESSAGE_TYPE_METHOD_CALL:
+                log_appendf(log,
+                            "MESSAGE_TYPE=method_call\n"
+                            "MESSAGE_PATH=%s\n"
+                            "MESSAGE_INTERFACE=%s\n"
+                            "MESSAGE_MEMBER=%s\n",
+                            message->metadata.fields.path ?: "<missing>",
+                            message->metadata.fields.interface ?: "<missing>",
+                            message->metadata.fields.member ?: "<missing>");
+                break;
+        case DBUS_MESSAGE_TYPE_SIGNAL:
+                log_appendf(log,
+                            "MESSAGE_TYPE=signal\n"
+                            "MESSAGE_PATH=%s\n"
+                            "MESSAGE_INTERFACE=%s\n"
+                            "MESSAGE_MEMBER=%s\n",
+                            message->metadata.fields.path ?: "<missing>",
+                            message->metadata.fields.interface ?: "<missing>",
+                            message->metadata.fields.member ?: "<missing>");
+                break;
+        case DBUS_MESSAGE_TYPE_METHOD_RETURN:
+                log_appendf(log,
+                            "MESSAGE_TYPE=method_return\n"
+                            "MESSAGE_REPLY_SERIAL=%d\n",
+                            message->metadata.fields.reply_serial);
+                break;
+        case DBUS_MESSAGE_TYPE_ERROR:
+                log_appendf(log,
+                            "MESSAGE_TYPE=method_return\n"
+                            "MESSAGE_ERROR_NAME=%s\n"
+                            "MESSAGE_REPLY_SERIAL=%d\n",
+                            message->metadata.fields.error_name,
+                            message->metadata.fields.reply_serial);
```

Comment:
I'd prefer not picking random fields in a global context. So far I used `DBUS_BROKER_*` prefix for fields we "invent". Furthermore, can we document them (at least mention them) in `src/util/log.c`?

Reply:
I added DBUS_BROKER_* prefix to everything now. However, is it not odd to document this in util/? Maybe it should be documented, but shouldn't util be generic? And wouldn't perhaps the manpage be the better place?

Reply:
Yeah, maybe we should rename `DBUS_BROKER_LOG_DROPPED` to something official, so util/log.c is dbus-broker-independent, and then move the docs to the wiki?

Reply:
Btw., what is wrong with just appending all **known** header-fields unconditionally?